src/test/bridges/erc4626/ERC4626.t.sol

```diff
@@ -0,0 +1,161 @@
+        fuzzerIgnoreList[0x7109709ECfa91a80626fF3989D68f67F5b1DD12D] = true; // HEVM
+        fuzzerIgnoreList[0x0000000000000000000000000000000000000003] = true; // precompile address
+    }
+
+    function testInvalidCaller(address _caller) public {
+        vm.assume(_caller != address(ROLLUP_PROCESSOR));
+        vm.prank(_caller);
+        vm.expectRevert(ErrorLib.InvalidCaller.selector);
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 0, address(0));
+    }
+
+    function testInvalidAuxData(uint64 _auxData) public {
+        vm.assume(_auxData > 1);
+        vm.expectRevert(ErrorLib.InvalidAuxData.selector);
+        vm.prank(address(ROLLUP_PROCESSOR));
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, _auxData, address(0));
+    }
+
+    function testListingNonERC4626Reverts(address _vault) public {
+        vm.assume(!fuzzerIgnoreList[_vault]);
+        vm.expectRevert();
+        bridge.listVault(_vault);
+    }
+
+    function testConvertRevertsWithoutApprovals() public {
+        // I wanted to fuzz the addresses here but there doesn't seem to be a cheat-code catching all reverts
+        // irrespective of revert message
+        vm.prank(address(ROLLUP_PROCESSOR));
+        vm.expectRevert("RDT:M:TRANSFER_FROM");
+        bridge.convert(assets[0], emptyAsset, shares[0], emptyAsset, 1 ether, 0, 0, address(0));
+    }
+
+    function testFullFlow(uint96 _assetAmount) public {
+        uint256 assetAmount = bound(_assetAmount, 10, type(uint256).max);
+
+        for (uint256 i = 0; i < shares.length; ++i) {
+            deal(assets[i].erc20Address, address(ROLLUP_PROCESSOR), assetAmount);
+
+            uint256 expectedAmount = IERC4626(shares[i].erc20Address).previewDeposit(assetAmount);
+
+            bridge.listVault(shares[i].erc20Address);
+
+            uint256 bridgeCallData = encodeBridgeCallData(id, assets[i], emptyAsset, shares[i], emptyAsset, 0);
+            (uint256 outputValueA, uint256 outputValueB, bool isAsync) = sendDefiRollup(bridgeCallData, assetAmount);
+
+            assertEq(outputValueA, expectedAmount, "Received amount of shares differs from the expected one");
+            assertEq(outputValueB, 0, "Non-zero outputValueB");
+            assertFalse(isAsync, "Bridge is not synchronous");
+
+            // Immediately redeem the shares
+            uint256 redeemAmount = outputValueA;
+
+            bridgeCallData = encodeBridgeCallData(id, shares[i], emptyAsset, assets[i], emptyAsset, 1);
+            (outputValueA, outputValueB, isAsync) = sendDefiRollup(bridgeCallData, redeemAmount);
+
+            assertApproxEqAbs(outputValueA, assetAmount, 2, "Received amount of asset differs from the expected one");
+            assertEq(outputValueB, 0, "Non-zero outputValueB");
+            assertFalse(isAsync, "Bridge is not synchronous");
+        }
+    }
+
+    function testEthWrappingAndUnwrapping(uint96 _assetAmount) public {
```

Comment:
Use the fuzzer for withdraw amount as well, make it bounded by the number of shares, but would be better to test that the values still in rollup after withdrawal is what is expected even if not exiting fully.

Reply:
Preferably, also alter the yield of the ERC4626 to have a different share value for withdraw than exit to properly test the accounting.

Reply:
Addressed in d4fccf3aaecaad00bab632c6af47fb737f646a55 and in e779d30da938fa4f63459af5ad0f942777f77f2f